EIPS/eip-6963.md

```diff
@@ -0,0 +1,230 @@
+
+Browser extensions are loaded in the web page and do not have a predictable pattern resulting in a race condition where the user is not in control to choose which wallet provider must be selected for exposing an Ethereum interface under the `window.ethereum` object.
+
+This results not only in a degraded user experience but also increases the barrier to entry for new browser extensions as users are forced to only install one browser extension at a time.
+
+Some browser extensions attempt to counteract this problem by delaying their injection to overwrite the same `window.ethereum` object which creates an unfair competition for wallet providers and lack of interoperability.
+
+In this proposal, we provide a solution that optimitizes for interoperability of multiple wallet providers and enables fairer competition by lowering the barrier to entry for new wallet providers to improve the user experience on Ethereum.
+
+This is achieved by introducing a set of window events to provide a two-way communication protocol between Ethereum libraries and injected scripts provided by browser extensions thus enabling users to select their wallet of choice.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Provider Info
+
+Each wallet provider will be exposed with the following interface `EIP6963ProviderInfo` that will be used to display to the user:
+
+```typescript
+/**
+ * Represents the assets needed to display a wallet
+ */
+interface EIP6963ProviderInfo {
+  walletId: string;
+  uuid: string;
+  name: string;
+  icon: string;
+}
+```
+
+The values in the `EIP6963ProviderInfo` MUST be used as follows:
+
+- walletId - globally unique identifier of the wallet provider (eg. `io.dopewallet.extension` or `awesomewallet`)
+- uuid - locally unique of the wallet provider (UUIDv4 compliant)
+- name - human-readable name of the wallet provider (e.g. `DopeWalletExtension` or `AwesomeWallet`)
+- icon - uri encoded image (RFC-3986 compliant, image must be squared with 96x96px minimum resolution)
+
+### Provider Detail
+
+The wallet provider will also expose their own [EIP-1193](./eip-1193.md) provider interface in parallel with the provider info provided above in the following interface `EIP6963ProviderDetail`
+
+```typescript
+interface EIP6963ProviderDetail {
+  info: EIP6963ProviderInfo;
+  provider: EIP1193Provider;
+}
+```
+
+The `EIP1193Provider` interface is documented at [EIP-1193](./eip-1193.md) and can be used to override the `window.ethereum` object once the user as explicitly selected it.
+
+### Window Event
+
+Different wallet providers might inject their scripts at different timeframes plus there is no guarantees that the Ethereum library in the web page will be loaded before all injected scripts are present in the page.
```

Comment:
This blurb probably belongs in the Rationale section, and not the specification.